frontend/src/pages/hardwareProfiles/manage/validationUtils.ts

```diff
@@ -0,0 +1,160 @@
+import { z } from 'zod';
+import { IdentifierResourceType, TolerationEffect, TolerationOperator } from '~/types';
+import {
+  validateDefaultCount,
+  validateMaxCount,
+  validateMinCount,
+} from '~/pages/hardwareProfiles/nodeResource/utils';
+import { splitValueUnit } from '~/utilities/valueUnits';
+import { HardwareProfileWarningType } from '~/concepts/hardwareProfiles/types';
+import {
+  createIdentifierWarningMessage,
+  determineIdentifierUnit,
+} from '~/pages/hardwareProfiles/utils';
+import { HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE } from '~/concepts/hardwareProfiles/const';
+import { hasCPUandMemory } from './ManageNodeResourceSection';
+
+const k8sNameRegex = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/;
+
+export const baseIdentifierSchema = z.object({
+  displayName: z.string().trim().min(1, 'Display name is required'),
+  identifier: z.string().trim().min(1, 'Identifier is required'),
+  resourceType: z.nativeEnum(IdentifierResourceType).optional(),
+  defaultCount: z.union([z.string(), z.number()]),
+  minCount: z.union([z.string(), z.number()]),
+  maxCount: z.union([z.string(), z.number()]).optional(),
+});
+
+export const identifierSchema = baseIdentifierSchema.superRefine((identifier, ctx) => {
+  const unitOptions = determineIdentifierUnit(identifier);
+
+  const checkNegative = (fieldName: string, value?: string | number) => {
+    if (value?.toString().startsWith('-')) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: createIdentifierWarningMessage(
+          `${fieldName} for ${
+            identifier.resourceType ?? identifier.displayName
+          } cannot be negative.`,
+        ),
+        path: [HardwareProfileWarningType.CANNOT_BE_NEGATIVE],
+      });
+      return true;
+    }
+    return false;
+  };
+
+  const checkDecimal = (value: number, fieldName: string) => {
+    if (!Number.isInteger(value)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: createIdentifierWarningMessage(
+          `${fieldName} for ${
+            identifier.resourceType ?? identifier.displayName
+          } cannot be a decimal.`,
+        ),
+        path: [HardwareProfileWarningType.CANNOT_BE_DECIMAL],
```

Comment:
path is not used like this. you should use param to add the cusotm type

```tsx
 params: { code: HardwareProfileWarningType.CANNOT_BE_DECIMAL },
```

Then path should be what specific item has an error. in this specific case it would be something like `["identifiers", index, "minValue" | "minValue" | "defaultValue"]`
I would pass that path in as part of check decimal

Reply:
do this for all the custom errors

Reply:
You are not using path now. we wont be able to assign the error to that specific field without it